contracts/SponsorFaucet.sol

```diff
@@ -45,36 +45,53 @@ contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
         if(_isAppliableForCollateral(dapp)) _applyForCollateral(dapp);
     }
 
+    function applyGas(address dapp) public nonReentrant whenNotPaused {
+        if(_isAppliableForGas(dapp)) _applyForGas(dapp);
+    }
+
+    function applyCol(address dapp) public nonReentrant whenNotPaused {
+        if(_isAppliableForCollateral(dapp)) 
+        _applyForCollateral(dapp);
+    }
+
     function isAppliable(address dapp) public returns (bool) {
         if(_isAppliableForGas(dapp) || _isAppliableForCollateral(dapp)) return true;
     }
 
     function _isAppliableForGas(address dapp) internal returns (bool) {
         require(address(this).balance >= gas_bound, "faucet out of money");
-        uint256 gas_balance = internal_sponsor_faucet.getSponsoredBalanceForGas(dapp);
+        uint256 gas_balance = internal_sponsor.getSponsoredBalanceForGas(dapp);
         require(gas_balance < gas_bound, "sponsored fund unused");
-        require(dapps[dapp].gas_amount < gas_total_limit, "over gas total limit");
+        require(dapps[dapp].gas_amount_accumulated < gas_total_limit, "over gas total limit");
         return true;
     }
 
     function _isAppliableForCollateral(address dapp) internal returns (bool) {
         require(address(this).balance >= collateral_bound, "faucet out of money");
-        uint256 collateral_balance = internal_sponsor_faucet.getSponsoredBalanceForCollateral(dapp);
+        uint256 collateral_balance = internal_sponsor.getSponsoredBalanceForCollateral(dapp);
         require(collateral_balance < collateral_bound, "sponsored fund unused");
-        require(dapps[dapp].collateral_amount < collateral_total_limit, "over collateral total limit");
+        require(dapps[dapp].collateral_amount_accumulated < collateral_total_limit, "over collateral total limit");
+        return true;
```

Comment:
If not appliable, then the cfx_call will get the error message in `require`, not a bool value.

Reply:
internal validate functions added